src/dropdown.ts

```diff
@@ -487,6 +478,45 @@ export default class GlideCoreDropdown
     ) {
       this.#inputElementRef.value.value = this.lastSelectedOption.label;
     }
+
+    const hasNoSelectedOptions = this.#optionElements.every(
+      ({ selected }) => !selected,
+    );
+
+    // When `value` is set on initial render, its setter is called before
+    // `connectedCallback()` and thus before the default slot has any assigned
+    // elements. So we set it again here after the initial render is complete
+    // so `this.#optionElements` isn't empty.
+    //
+    // Additionally, `#onDefaultSlotChange()` is called after `firstUpdated()`
+    // and sets `value` based on which options are selected. And the initial `value`
+    // may conflict with the one derived from which options are selected.
+    //
+    // So we have a decision to make. On first render, do we defer to the initial
+    // `value` and select and deselect options below? Or do we defer to `#onDefaultSlotChange()`
+    // and let that method change `value` from its initial value based on which options
+    // are selected?
+    //
+    // It's largely a toss-up. But the latter seems like the logial choice given
+    // `#onDefaultSlotChange()` is called after `firstUpdated()`. In other words, we
+    // defer to the lifecycle. `#onDefaultSlotChange()` is called second. So it gets to
+    // override what `value` was initially.
+    //
+    // If no options are selected, then it's obvious that the consumer's intention is
+    // to select options based on the initial `value` and that the initial `value` is
+    // the intended one. So we proceed.
```

Comment:
Hard one to explain. Let me know if this doesn't make sense.

Reply:
Makes sense to me 👍